### PR TITLE
add libvirt check in autoconf

### PR DESCRIPTION
**Pull request by gao-feng**
Signed-off-by: Gao feng <omarapazanadi@gmail.com>

### CODE REVIEW

**Comment by gnawux** [comment]
will it compile fail if no libvirt present? If so, we need some code to disable libvirt related code if libvirt develop files are not available.

**Comment by gao-feng** [comment]
--without-libvirt, same with xen

File: configure.ac [hunk 1]
```diff
@@ -72,6 +72,20 @@ AC_TYPE_UINT8_T
...
+LIBVIRT_REQUIRED="1.2.2"
+
+AC_ARG_WITH([libvirt],
+            [AS_HELP_STRING([--without-libvirt],
+                            [run hyperd with libvirt])],
+            [with_libvirt=no],[with_libvirt=yes])
+
+if test "$with_libvirt" = yes; then
+	PKG_CHECK_MODULES([libvirt], [libvirt >= $LIBVIRT_REQUIRED], [], [
+		AC_MSG_ERROR([libvirt-devel >= $LIBVIRT_REQUIRED is required or reconfigure runv with --without-libvirt])])
```
Comment (laijs) [inline configure.ac]: "reconfigure runv with", it should be hyper.

Comment (laijs) [inline configure.ac]: it should not fail when there is no libvirt-dev.

Comment (laijs) [inline configure.ac]: --with-xen will be silent turn off when there is no xen-dev installed. I think it should complain and fail.

--with-xen & xen-dev  => xen
--with-xen & no xen-dev  =>  fail
--without-xen & xen-dev  => no xen
--without-xen & no xen-dev  => no xen
xen-dev  => xen
no xen-dev  => no xen

the same for libvirt

Comment (gao-feng) [inline configure.ac]: it won't fail, if there is no libvirt-dev or the version is incorrect, you should reconfigure hyper/runv with --without-libvirt, please check https://github.com/hyperhq/runv/pull/96.

We want xen&libvirt driver being available by default, so we outthrow warning if there is no xen or libvirt available,and user can use --without- option to reconfigure hyper/runv.

**Pull request merged**